src/main/java/io/nats/client/impl/NatsJetStream.java

```diff
@@ -454,11 +458,12 @@ public List<String> getChanges(ConsumerConfiguration serverCc) {
             record(rateLimit != null && !rateLimit.equals(serverCcc.getStartSequence()), "rateLimit", changes);
 
             // MaxDeliver is a special case because -1 and 0 are unset where other unsigned -1 is unset
-            record(LongChangeHelper.MAX_DELIVER.wouldBeChange(maxDeliver, serverCcc.maxDeliver), "maxDeliver", changes);
+            record(IntegerChangeHelper.MAX_DELIVER.wouldBeChange(maxDeliver, serverCcc.maxDeliver), "maxDeliver", changes);
 
-            record(maxAckPending != null && !maxAckPending.equals(serverCcc.getMaxAckPending()), "maxAckPending", changes);
-            record(maxPullWaiting != null && !maxPullWaiting.equals(serverCcc.getMaxPullWaiting()), "maxPullWaiting", changes);
-            record(maxBatch != null && !maxBatch.equals(serverCcc.getMaxBatch()), "maxBatch", changes);
+            record(maxAckPending != null && !maxAckPending.equals((int)serverCcc.getMaxAckPending()), "maxAckPending", changes);
+            record(maxPullWaiting != null && !maxPullWaiting.equals((int)serverCcc.getMaxPullWaiting()), "maxPullWaiting", changes);
+            record(maxBatch != null && !maxBatch.equals(serverCcc.getMaxBatch().intValue()), "maxBatch", changes);
+            record(maxBytes != null && !maxBytes.equals(serverCcc.getMaxBytes()), "maxBytes", changes);
```

Comment:
casting to int is required because an Integer can never be equal to a Long object, and the getters represent integers anyway even though their return type was not changed, avoiding backward compatibility problems.